src/middlewared/middlewared/plugins/system.py

```diff
@@ -1626,12 +1626,13 @@ async def setup(middleware):
     await middleware.call('system.general.set_language')
     await middleware.call('system.general.set_crash_reporting')
 
-    asyncio.ensure_future(middleware.call('system.advanced.autotune', 'sysctl'))
+    if osc.IS_FREEBSD:
+        asyncio.ensure_future(middleware.call('system.advanced.autotune', 'sysctl'))
 
     if sysctl:
         await update_timeout_value(middleware)
 
-    for srv in ['initshutdownscript', 'tunable', 'vm']:
+    for srv in (['initshutdownscript', 'tunable', 'vm'] if osc.IS_FREEBSD else []):
```

Comment:
Why?

Reply:
FreeBSD imposed a limit on shutdown defaulting to 120 secs, so even if we had tasks running, that was enforced and middleware was forcibly killed. So this hook is specific to FreeBSD and that's why this change has been made

Reply:
Put entire loop under this condition then so it is clear